lib/util/cmd/logger.go

```diff
@@ -12,37 +12,25 @@
 // See the License for the specific language governing permissions and
 // limitations under the License.
 
-package logger
+package cmd
 
 import (
-	"sync"
-
 	"github.com/pingcap/TiProxy/lib/config"
-	"github.com/pingcap/TiProxy/lib/util/cmd"
 	"github.com/pingcap/log"
 	"go.uber.org/zap"
 	"go.uber.org/zap/zapcore"
 )
 
-var registerEncoders sync.Once
-
 func buildEncoder(cfg *config.Log) (zapcore.Encoder, error) {
-	zapcfg := zap.NewDevelopmentConfig()
-	zapcfg.Encoding = cfg.Encoder
-	zapcfg.DisableStacktrace = true
-	encoder, err := log.NewTextEncoder(&log.Config{})
-	if err != nil {
-		return nil, err
+	encfg := zap.NewProductionEncoderConfig()
```

Comment:
NewProductionEncoderConfig enables sampling and it doesn't set DisableStacktrace to true.

Reply:
Note that I am only using `ProductionEncoderConfig` instead of `ProductionConfig()` or `zapcfg.Build()`. `Stacktraces` are handled in your `BuildLogger` function.

The config is exclusively used by encoder only.